subsys/event_manager_proxy/event_manager_proxy.c

```diff
@@ -0,0 +1,671 @@
+	ret = k_sem_take(&ipc->rsp_ready, K_NO_WAIT);
+	if (ret == 0) {
+		LOG_ERR("Response received ready during response submitting");
+	}
+
+	ipc->rsp.id = id;
+	ipc->rsp.res = res;
+
+	ret = k_work_submit(&ipc->rsp_work);
+	return ret;
+}
+
+/**
+ * @brief The endpoint bound callback
+ *
+ * This callback is called when the endpoint is bonded on the remote core.
+ * When this is received the communication between cores is possible.
+ *
+ * @param priv The pointer of the related element of the @ref emp_ipc_data array.
+ */
+static void ipc_ept_bound(void *priv)
+{
+	struct emp_ipc_data *ipc = priv;
+
+	k_event_set(&(ipc->bonded), 0x1);
+}
+
+/**
+ * @brief The data received on endpoint callback
+ *
+ * This callback is called when there is some data received from the remote core.
+ *
+ * @param data The pointer to the data received.
+ * @param len  The length of the data received.
+ * @param priv The pointer of the related element of the @ref emp_ipc_data array.
+ */
+static void ipc_ept_recv(const void *data, size_t len, void *priv)
+{
+	struct emp_ipc_data *ipc = priv;
+
+	/* We are expecting this callback to be called on the thread context */
+	__ASSERT_NO_MSG(!k_is_in_isr());
+
+	if (ipc->started && emp_started) {
+		/* Incoming event data - copy and submit */
+		void *ev = event_manager_alloc(len);
+
+		memcpy(ev, data, len);
+		_event_submit(ev);
+	} else {
+		/* Incoming command - execute */
+		if (len < sizeof(struct emp_cmd)) {
+			LOG_ERR("Unexpected command size received: %u", len);
+			__ASSERT_NO_MSG(false);
+		}
+		switch (((struct emp_cmd *)data)->cmd) {
+		case EMP_CMD_REGISTER:
+		{
+			/* Only command responses are accepted after the interfacace is started */
+			__ASSERT_NO_MSG(!ipc->started);
+			/* Expecting valid command header and at least 1 name character */
+			if (len < (sizeof(struct emp_cmd_register) + 2)) {
+				LOG_ERR("Unexpected register command size (%u)", len);
+				__ASSERT_NO_MSG(false);
+			}
+			const struct emp_cmd_register *cmd = data;
+			struct event_type *et = find_event_by_name(cmd->name);
+			int ret = 0;
+
+			if (!et) {
+				LOG_ERR("Cannot find requested event: \"%s\"",
+					log_strdup(cmd->name));
+				ret = -ENOENT;
+			} else {
+				size_t context_idx = ept2idx(priv);
+				size_t ev_idx = ev2idx(et);
+
+				event_manager_proxy_array[ev_idx].event[context_idx] = cmd->id;
+				LOG_DBG("Remote event registered for \"%s\" -> ipc %d",
+					log_strdup(cmd->name),
+					context_idx);
+			}
+			ret = submit_rsp(ipc, cmd->id, ret);
+			__ASSERT_NO_MSG(ret);
+			break;
+		}
+		case EMP_CMD_START:
+		{
+			/* Only command responses are accepted after the interfacace is started */
+			__ASSERT_NO_MSG(!ipc->started);
+			unsigned int n;
+			bool all_started;
+
+			ipc->started = true;
+			LOG_DBG("Event transmission on ipc %d started", ept2idx(priv));
+			/* Check if all remote cores are started and mark the fact */
+			for (n = 0, all_started = true; n < ARRAY_SIZE(emp_ipc_data); ++n) {
```

Comment:
I have changed `n` to `i` to maintain the consistency among most places. I know this is not very important but `i` really comes into mind automatically

Reply:
In my mind - i for integers, n for unsigned integers.